### ISSUE TITLE
question: Support OCI artifacts with artifactType

### ISSUE THREAD

**Issue by knqyf263** [google/go-containerregistry#1832]
I tried to generate a manifest with `config.mediaType: "application/vnd.oci.empty.v1+json"` and `artifactType: "application/vnd.example+type"` as described below.
https://github.com/opencontainers/image-spec/blob/93f6e65855a1e046b42afbad0ad129a3d44dc971/manifest.md#guidelines-for-artifact-usage

Especially, I want to generate the second example.

```
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.example+type",
  "config": {
    "mediaType": "application/vnd.oci.empty.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.example+type",
      "digest": "sha256:e258d248fda94c63753607f7c4494ee0fcbe92f1a76bfdac795c9d84101eb317",
      "size": 1234
    }
  ]
}
```

I used [empty.Image](https://github.com/google/go-containerregistry/blob/dbcd01c402b2f05bcf6fb988014c5f37e9b13559/pkg/v1/empty/image.go#L26) and the mutate package, like [mutate.ConfigMediaType](https://github.com/google/go-containerregistry/blob/dbcd01c402b2f05bcf6fb988014c5f37e9b13559/pkg/v1/mutate/mutate.go#L542).

```
content := []byte(`hello`)
artifactType := "application/vnd.example+type"
img, err := mutate.AppendLayers(empty.Image, static.NewLayer(content, artifactType))
if err != nil {
	panic(err)
}

img = mutate.ConfigMediaType(img, artifactType)
```

However, it generates the following manifest.

```
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
  "config": {
    "mediaType": "application/vnd.example+type",
    "size": 233,
    "digest": "sha256:185de42e4ed8eb4fc529c9a3ea02d808c1dfe4466737c0bcf30a5d7ad855088f"
  },
  ...
```

`mutate.ConfigMediaType(img, "application/vnd.oci.empty.v1+json")` also doesn't work.

Am I missing something? I didn't find `artifactType` in [the manifest definition](https://github.com/google/go-containerregistry/blob/dbcd01c402b2f05bcf6fb988014c5f37e9b13559/pkg/v1/manifest.go#L24-L32). Are you intentionally not supporting OCI artifacts as OCI 1.1 is still in progress?

**Comment by wy65701436** [comment]
hi, I'm experiencing the same problem when building an OCI v1.1 manifest. It seems that there is no way to specify the artifactType. Is this still planned?

**Comment by malt3** [comment]
This is now implemented as part of #2269. Can we reopen and close as done?
(@Subserial gentle ping. I think you are the current maintainer)
